src/rgw/rgw_rest_pubsub.cc

```diff
@@ -163,14 +242,18 @@ class RGWPSCreateTopicOp : public RGWOp {
 };
 
 void RGWPSCreateTopicOp::execute(optional_yield y) {
-  op_ret = get_params();
-  if (op_ret < 0) {
-    return;
+  if (!dest.push_endpoint.empty() && dest.persistent) {
```

Comment:
great! this is the right place

Reply:
> great! this is the right place

yep this and same for the DeleteTopic, we were deleting in get_params()